[CS2113T-W11-1] Kolinux

saravanamani1999 opened the pull request:
Kolinux is built to help NUS Computer Engineering freshmen to better integrate into university life by allowing them to manage their work and learn more about NUS, all in 1 platform. It is optimized for CLI users so that they can access the information that they require faster by typing in commands.

fansxx commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+### Before writing code
+
+1. Configure the code style
+   * Ensure that your coding style matches our coding style
```
'our coding style' is not clear, perhaps you can have a hyperlink to the coding standards you follow?

fansxx commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+
+The class diagram below model the associations within the `module` component
+
+![Module Class Diagram](assets/images/ModuleClassDiagram.png)
```
![image](https://user-images.githubusercontent.com/70215507/139004235-0aacfc41-bc4b-47eb-89cd-7b2c9a71c682.png)
the '1' multiplicity for `ModuleList` is not very obvious. Try moving it to the white space if PlantUML allows.

JaCh23 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+
+The class diagram below describes the `commands` component.
+
+❕ _Note: XYZCommand in this diagram represents HelpCommand, ExitCommand, and InvalidCommand._
```
Is it a better idea to write out HelpCommand, ExitCommand, and InvalidCommand instead of a generic "XYZCommand"?

fansxx commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+The sequence diagram below shows a high-level overview of the interaction between entities during the execution
+of a user input _(XYZCommand represents any class that inherits from Command)_.
+
+![Overview Sequence Diagram](assets/images/OverviewSequenceDiagram.png)
```
![image](https://user-images.githubusercontent.com/70215507/139004714-e5cc57db-380d-405e-a7bc-35c5974e2995.png)
the `Kolinux` entity does not have an activation bar. is it supposed to have an activation bar as the other entities have activation bars, for the cohesiveness of the diagram?

JaCh23 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+
+![Commands Class Diagram](assets/images/CommandsClassDiagram.png)
+
+All `..Command` inherit from the abstract `Command` class, which has interactions with `KolinuxLogger` so that
```
Should `..Command` be `:Command` instead?

fansxx commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+
+Example: `myModules` is initialized with single `ModuleDetails` instance corresponding to `CS2113T`
+
+<img src="assets/images/moduleListInit.png" width="550">
```
![image](https://user-images.githubusercontent.com/70215507/139005974-1713a528-443e-40af-9348-902cf8bf224c.png)
if the `CS2113T` is a `String` object, should there be " " around it, like `"CS2113T"`?

JaCh23 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+
+The class diagram below describes the interaction between `CapCalculator` and its subclasses.
+
+![CapCalculator Class Diagram](assets/images/CapCalculatorClassDiagram.png)
```
Is {abstract} necessary?

![image](https://user-images.githubusercontent.com/24263853/139006122-e4b8f27f-9b21-408a-b381-8bdbb0325345.png)

fansxx reviewed (commented):
good work so far! no major bugs, just minor typos:)

JaCh23 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+* Lesson class is inherited by `Tutorial`, `Lecture` and `Lab` to add lessons based on the `LESSON_TYPE` as shown 
+in the example below.
+
+Given below are the examples of the usage of `timetable add` of lessons to the timetable.
+
+Example 1: Adding a tutorial to the `lessonStorage` ( lesson of type `TUT` )
+
+![Example 1](assets/images/timetableAdd1.png)
+
+Example 2: Adding a lecture to the `lessonStorage` ( lesson of type `LEC` )
+
+![Example 2](assets/images/timetableAdd2.png)
+
+Example 3: Adding a lab to the `lessonStorage` ( lesson of type `LAB` )
+
+![Example 3](assets/images/timetableAdd3.png)
+
+The following sequence diagrams shows the `timetable add` mechanism:
```
Are there too many examples that may over-complicate things? Perhaps cutting down to 1 or 2 diagrams is a better idea?

JaCh23 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+* The following diagram shows the sequence of parsing the user input and executing `TimetableCommand#addLesson()`
+for the `timetbale add` command
+
+![Sequence Diagram1](assets/images/TimetableAddSequenceDiagram1.png)
```
Perhaps the words are too small, maybe splitting into 2 separate diagrams makes the diagram larger and clearer?

![image](https://user-images.githubusercontent.com/24263853/139006831-971df01e-7eef-48c4-bb99-197f2fbe13c8.png)

JaCh23 reviewed (commented):
Very detailed DG, but some details could be an overkill aka can consider simplifying, else looks very good!

lezongmun commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+
+Example 1: Adding a tutorial to the `lessonStorage` ( lesson of type `TUT` )
+
+![Example 1](assets/images/timetableAdd1.png)
+
+Example 2: Adding a lecture to the `lessonStorage` ( lesson of type `LEC` )
+
+![Example 2](assets/images/timetableAdd2.png)
+
+Example 3: Adding a lab to the `lessonStorage` ( lesson of type `LAB` )
+
+![Example 3](assets/images/timetableAdd3.png)
```
Perhaps these examples might be unnecessarily repetitive? Feels like there are too many sample inputs and outputs, which might not add much value. Like @JaCh23 mentioned, maybe its better to cut down the diagrams.

lezongmun commented on docs/assets/plantuml/PlannerAddSequenceDiagram2.puml:
```diff
@@ -0,0 +1,68 @@
+
+end
+end
+end
```
![image](https://user-images.githubusercontent.com/61373716/139170374-aaa7c5db-2bb5-4b32-ab22-785bb677c5f5.png)
If it's possible, would it better to use some background colours for the frame headers, to differentiate the frames clearer?

lezongmun commented on docs/assets/plantuml/PlannerAddSequenceDiagram2.puml:
```diff
@@ -0,0 +1,68 @@
+planner --> planner : hasConflict
+deactivate planner
+
+opt hasTimeConflict && !allowConflict
```
Would it be better to phrase the condition for optional path to something like `has disallowed time conflict`, so that it is is similar to the phrasing of your other conditions in the DG?

lezongmun commented on docs/assets/plantuml/ModuleStoreSequenceDiagram.puml:
```diff
@@ -0,0 +1,66 @@
+
+mlc-->[ : result
+deactivate mlc
+
```
![image](https://user-images.githubusercontent.com/61373716/139172138-c5f811c4-c7ec-4519-9990-f71b5d1427ff.png)
Would it be better to include variable types in the diagram for the functions inputs and return values?  I see that in some cases they are included, so perhaps it is better to be consistent.
For example, `mlc -> res: new CommandResult(message : String)` or `command: Command`

lezongmun reviewed (commented):
Great job for the Developer Guide, no major issues that I can see

nguyin commented on docs/assets/plantuml/PlannerClassDiagram.puml:
```diff
@@ -0,0 +1,82 @@
+@startuml
```
Are these diagram necessary? It feels a bit too repetitive as the changes made to each diagram is very minor. Maybe it is possible to just incorporate all of this into one diagram?
<img width="941" alt="Screenshot 2021-10-28 at 4 20 02 PM" src="https://user-images.githubusercontent.com/70097982/139216471-094c6b24-b20b-40a8-9ffd-625ba679a946.png">
<img width="901" alt="Screenshot 2021-10-28 at 4 20 39 PM" src="https://user-images.githubusercontent.com/70097982/139216494-d6ab86c1-da49-4502-a185-654125c6c784.png">

nguyin commented on docs/assets/plantuml/ArchitectureDiagram.puml:
```diff
@@ -0,0 +1,58 @@
+@startuml
+'https://plantuml.com/component-diagram
+
+!include Style.puml
```
This diagram feels a bit repetitive. Is it meant to show the different interactions (association and dependency) between the objects? There are three filteredPlanner created objects, maybe it is possible to make the diagram less repetitive?
<img width="870" alt="Screenshot 2021-10-28 at 4 27 09 PM" src="https://user-images.githubusercontent.com/70097982/139217613-9ca27f84-84c7-4e63-898b-0b84e29269f7.png">

nguyin commented on docs/assets/plantuml/ArchitectureDiagram.puml:
```diff
@@ -0,0 +1,58 @@
+component Main MAIN_COLOR
+component Kolinux KOLINUX_COLOR
+
+package util UTIL_COLOR {
```
The two steps seem quite similar, though you are creating two instances of an object. Maybe can just have one diagram to depict this as it is quite repetitive also?
<img width="962" alt="Screenshot 2021-10-28 at 4 33 27 PM" src="https://user-images.githubusercontent.com/70097982/139218726-c7758948-2710-4678-a911-67682eb0a75a.png">

teoziyiivy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+2. Extend the functionality of _Kolinux_
+
+This guide will bring you through the [overall design](#design) of _Kolinux_, the various 
+[implementations](#implementation) and their mechanisms. We have also provided insights into our target users
```
As you have hyperlinked overall design and implementations, maybe you can hyperlink the insights of target users section as well?

EdwardZYWang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+
+     
+
+4. 
```
Is there supposed to be something after point 4.?

EdwardZYWang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+* `Route#getBusStopNumber()` - Converts the user given bus stop names to bus stop numbers which can then be used to find if the bus stops are connected.
+* `Route#checkDirectRoutes(ArrayList<String> busNumbers)` - Check whether there is a direct bus route between the 2 user given bus stops by calling the `Graph#isConnected(int u, int v)` method which uses BFS to check if any 2 points in the directed unweighted graph are connected.
+* `Route#checkIndirectRoutes(ArrayList<String> busOne, ArrayList<String> busTwo, ArrayList<String> midLoc)` - Checks whether there is an alternate route between the 2 user given bus stops which requires a single change of bus at an intermediate bus stop.
+
```
perhaps there is a better way to format the points, maybe leave spacings between each points or change font colours of the keywords

EdwardZYWang replied:
![image](https://user-images.githubusercontent.com/69493822/139385725-e54671ca-4e18-48e0-a6a9-f0935d5474b4.png)

EdwardZYWang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+functionality.
+
+* `CapCalculator#executeCommand()` — which is an overridden method from `Command` is used to facilitate cap calculation
+and exception handling methods.
+* `CapCalculator#checkModulesNotEmpty()` — which ensures that the module list of the object is not empty.
+* `CapCalculator#checkInvalidModules()` — which checks if there are any invalid modules after the cap calculation.
+
+Below is the sequence diagrams showing important steps of how `cap code` operates:
+
+![Cap Code Sequence Diagram 1](assets/images/capCodeSeq1.png)
+
+![Cap Code Sequence Diagram 2](assets/images/capCodeSeq2.png)
+
+![Cap Code Sequence Diagram 3](assets/images/capCodeSeq3.png)
+
+### Bus routes feature
+The bus routes feature is facilitated by the `BusRouteCommand` class. The `BusRouteCommand` class extends the `Command` class. 
+When the user invokes and uses the bus routes feature the `BusRouteCommand` constructor creates a `Route` class object and passes
+the `input` string to the `Route` class. The operation is implemented in the following way.
+
+* The overriden function `executeCommand()` calls the `Route#checkRoutes()` method. 
+* The `Route#checkRoutes()` contains the `Route#getBusStopNumber()`, `Route#checkDirectRoutes(ArrayList<String> busNumbers)` and `Route#checkIndirectRoutes(ArrayList<String> busOne, ArrayList<String> busTwo, ArrayList<String> midLoc)` - Checks whether there is a direct or an indirect route between the 2 user given bus stops and returns a string depending on the result.
+* `Route#getBusStopNumber()` - Converts the user given bus stop names to bus stop numbers which can then be used to find if the bus stops are connected.
+* `Route#checkDirectRoutes(ArrayList<String> busNumbers)` - Check whether there is a direct bus route between the 2 user given bus stops by calling the `Graph#isConnected(int u, int v)` method which uses BFS to check if any 2 points in the directed unweighted graph are connected.
+* `Route#checkIndirectRoutes(ArrayList<String> busOne, ArrayList<String> busTwo, ArrayList<String> midLoc)` - Checks whether there is an alternate route between the 2 user given bus stops which requires a single change of bus at an intermediate bus stop.
+
+The following sequence diagram explains the bus routes feature.
+
+![sequenceDiagram](assets/images/BusRouteSequenceDiagram.png)
 
 ## Product scope
-### Target user profile
+### Target user profile:
 
-{Describe the target user profile}
+* needs help with adapting to university life by understanding the grading system, university bus routes etc.
+* has a need to manage their schedule along with the modules they are taking in the semester
+* can type fast
+* prefers typing to mouse interactions
+* is reasonably comfortable using CLI apps
```
starting letters of points should be capital letters

teoziyiivy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+2. Configure JDK: Ensure your IDE is configured to JDK 11
+3. Import the project as a Gradle Project 
+4. Verify the setup by running `seedu.Kolinux.Main`
+   1. Try out a few commands and ensure they're working properly
```
Could hyperlink the manual testing part which is at the bottom of the UG?

EdwardZYWang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+
+The figures below represent the sequence diagrams of the Add to Planner mechanism:
+
+![Planner Sequence Diagram 1](assets/images/PlannerAddSequenceDiagram1.png)
```
Maybe the words are too small, maybe splitting into 2 separate diagrams makes the diagram larger and clearer?
![image](https://user-images.githubusercontent.com/69493822/139386400-e9c6b0bb-f263-42b3-8de6-5f6bed072a55.png)

teoziyiivy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
+
+
+
+### CAP Calculator by module code feature
```
Is there any reason this wasnt presented step by step using the diagrams like the storing/deleting of a module feature above and you guys chose to lump the diagrams at the end?

teoziyiivy commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,535 @@
 
 ## Non-Functional Requirements
 
 {Give non-functional requirements}
```
I think you guys forgot to remove this line?

powzx committed:
change color for bus route seq diag

nonrnp committed:
Add option to reset module grade using "module grade" command

powzx committed:
change color for routes class diag

nonrnp committed:
Edit text ui expected output

powzx committed:
Merge pull request #274 from NonRNP/branch-BugFixing

Add option to reset module grade

powzx committed:
improve readability in filterPlanner()

powzx committed:
add comment for Parser

powzx committed:
add specific feedback for event add

powzx committed:
improve event conflict message

powzx committed:
Refactor deleteEvent

show specific msg when deleting event

giterator committed:
Edit wording of module component description

powzx committed:
update user guide with new messages

giterator committed:
Make Command class abstract in module component diagram

powzx committed:
Update planner class diagram with new method signature

powzx committed:
Merge branch 'master' into branch-Polish

giterator committed:
Improve code quality

giterator committed:
Refactor RESET and 0 from ModuleDetails and ModuleList to constants

giterator committed:
Remove unused imports from JsonReader, use RESET_GRADE constant

nonrnp committed:
Improve cap command message when no calculator type is given

saravanamani1999 committed:
fix json special character error

saravanamani1999 committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-Polish
Merge with upstream

saravanamani1999 committed:
Merge pull request #275 from powzx/branch-Polish

Add event specific messages for add and delete

saravanamani1999 committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-Polish
Merge with upstream

giterator committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into review

giterator committed:
Merge pull request #276 from saravanamani1999/branch-Polish

Fix unicode apostrophe in json file

giterator committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into review

giterator committed:
Modify json

nonrnp committed:
Edit cap output message

nonrnp committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-BugFixing

nonrnp committed:
Update cap output message in UG, text ui test

powzx committed:
Merge pull request #277 from NonRNP/branch-BugFixing

Edit output message from cap calculator

giterator committed:
Change module store to module add

giterator committed:
Refactor storeModuleByCode to addModuleByCode, edit DG

giterator committed:
Edit UG to reflect module add

giterator committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into review

powzx committed:
Merge pull request #278 from giterator/review

Review edits

nonrnp committed:
Edit cap calculator message, separate "module cap" and "module cap desired_cap" command in ModuleCommand message

nonrnp committed:
More options of non calculating grades

shyamgj1900 committed:
Update PPP

shyamgj1900 committed:
Merge branch 'master' into master

powzx committed:
Merge pull request #280 from shyamgj1900/master

Update PPP

giterator committed:
Convert all unicode characters read from NUSmods JSON to ASCII

giterator committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into review

powzx committed:
Merge pull request #281 from giterator/review

Convert all unicode characters read from NUSmods JSON to ASCII

shyamgj1900 committed:
Merge branch 'master' of https://github.com/shyamgj1900/tp

giterator committed:
Create PPP for giterator

shyamgj1900 committed:
Merge pull request #284 from giterator/review

Create PPP for giterator

giterator committed:
Refactor all grade magic literals as constants in Grade class

giterator committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into review

nonrnp committed:
Merge pull request #287 from giterator/review

Refactor all grade magic literals as constants in Grade class

nonrnp committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-BugFixing

# Conflicts:
#	src/main/java/seedu/kolinux/capcalculator/CapCalculatorByCode.java
#	src/main/java/seedu/kolinux/commands/ModuleCommand.java
#	src/main/java/seedu/kolinux/module/ModuleDetails.java

powzx committed:
Merge pull request #288 from NonRNP/branch-BugFixing

Add more grade option

powzx committed:
1000th commit

powzx committed:
shorten unnecessary lines in planner diagram

saravanamani1999 committed:
Merge pull request #289 from powzx/branch-Final

Shorten diagram and minor edits to DG

powzx committed:
lose weight for all sequence diagrams

powzx committed:
resolve javadoc bug

powzx committed:
Merge branch 'master' into branch-Final

powzx committed:
resolve reposense link bug in our ppp

powzx committed:
edit bus route sd

saravanamani1999 committed:
Merge pull request #290 from powzx/branch-Final

Resolve minor bugs and adjust diagram size

saravanamani1999 committed:
Update UG with limitations and add FAQs

saravanamani1999 committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-Polish
Merge with upstream

powzx committed:
Merge pull request #291 from saravanamani1999/branch-Polish

Update UG with limitations and add FAQs

saravanamani1999 committed:
implement timetable list for more concise viewing of timetable data

saravanamani1999 committed:
update UG and help command with timetable list

saravanamani1999 committed:
fix checkstyle error

saravanamani1999 committed:
gradle errors

saravanamani1999 committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-Polish
Merge with upstream

shyamgj1900 committed:
Merge pull request #293 from saravanamani1999/branch-Polish

Implemented `timetable list DAY`

nonrnp committed:
add error message for module grade when input grade is the same as stored grade

powzx committed:
Merge pull request #294 from NonRNP/branch-BugFixing

Add error message for module grade when input grade is the same as stored grade

powzx committed:
Merge branch 'master' into branch-Final

nonrnp committed:
Add PPP

nonrnp committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-UGDG

nonrnp committed:
Edit  reposense link

powzx committed:
fix some javadoc inconsistencies in general classes

powzx committed:
adjust indentation and add javadoc for handleClearCommand()

powzx committed:
Merge pull request #295 from NonRNP/branch-UGDG

Add nonrnp PPP

powzx committed:
refactor and remove unused import

powzx committed:
add infinite loop to handle prompt replies

powzx committed:
slap myself harder until i fly

powzx committed:
update javadoc

powzx committed:
update javadoc

powzx committed:
Planner and Prompt design revamp

powzx committed:
Merge branch 'master' into branch-Final

saravanamani1999 committed:
Merge pull request #296 from powzx/branch-Final

Planner design revamp

powzx committed:
Merge branch 'master' into branch-Final

powzx committed:
Various diagram and DG updates according to design revamp

saravanamani1999 committed:
Merge pull request #297 from powzx/branch-Final

DG and diagrams update according to updated design

saravanamani1999 committed:
add prompt for exceeding workload in timetable

powzx committed:
Merge branch 'master' into branch-Final

saravanamani1999 committed:
Make code slap more

powzx committed:
Add more junit tests for planner

powzx committed:
refine javadoc

saravanamani1999 committed:
add more junit tests for list and timetable prompt

powzx committed:
correct any outdated javadoc

powzx committed:
add another junit

powzx committed:
resolve file reading bug

if planner was corrupted then modulelist and timetable will not be initialized but it's fixed now

nonrnp committed:
Merge pull request #298 from powzx/branch-Final

Add more JUnit tests for planner to increase coverage and bug fix

powzx committed:
Merge branch 'master' into branch-Final

powzx committed:
update powzx ppp

powzx committed:
small correction in dg

powzx committed:
add junit for PromptHandler

powzx committed:
fix checkstyletest

powzx committed:
add integration testing in planner with modulelist

saravanamani1999 committed:
Merge pull request #299 from powzx/branch-Final

Add integration testing to planner with modulelist, minor corrections in docs and ppp

saravanamani1999 committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-Polish

saravanamani1999 committed:
add javadoc for all timetable methods

saravanamani1999 committed:
fix test cases in timetable

powzx committed:
Merge pull request #300 from saravanamani1999/branch-Polish

Added prompt for `timetable add`

powzx committed:
Merge branch 'master' into branch-Final

saravanamani1999 committed:
update ug command summary table

saravanamani1999 committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-Polish
Merge with upstream

saravanamani1999 committed:
update ug help expected output with timetable list

powzx committed:
Merge pull request #301 from saravanamani1999/branch-Polish

updated ug command summary table

powzx committed:
Merge branch 'master' into branch-Final

nonrnp committed:
Modules that cannot be su no longer accept su grade

nonrnp committed:
check su grade validity in cap calculator

nonrnp committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-BugFixing

nonrnp committed:
Improve JUnit test for cap calculator by code

nonrnp committed:
Update cap calculator diagrams in DG

nonrnp committed:
Edit PPP

nonrnp committed:
Fix code style

nonrnp committed:
Edit text ui test

giterator committed:
Merge pull request #302 from NonRNP/branch-BugFixing

Prevent user from assigning su grade to non su-able modules

powzx committed:
abstract date and time checking to parser

add ability to view day of week when u list events

nonrnp committed:
CS/CU mods now accept only cs/cu grade

nonrnp committed:
Forbid assigning normal mods with cs/cu grades

nonrnp committed:
Fix junit test

powzx committed:
write test cases for more specific exceptions thrown

powzx committed:
update ug with new expected output

powzx committed:
Merge pull request #303 from NonRNP/branch-BugFixing

CS/CU mods now accept only cs/cu grade

powzx committed:
Merge branch 'master' into branch-Final

shyamgj1900 committed:
Merge branch 'AY2122S1-CS2113T-W11-1:master' into master

nonrnp committed:
Merge pull request #304 from powzx/branch-Final

Abstract verification of datetime into parser, add more specific exceptions

powzx committed:
add instructions to manual test planner in DG

powzx committed:
update glossary

powzx committed:
Move all puml files to docs/diagrams

website says so

powzx committed:
Merge branch 'master' into branch-Final

powzx committed:
Refactor isLesson to isIntegratedEvent

this flag will be set if it is constructed from either a lesson or exam

shyamgj1900 committed:
Update DG

Add instructions for manual testing

shyamgj1900 committed:
Merge branch 'master' into master

nonrnp committed:
Edit module details toString message

powzx committed:
Merge pull request #306 from shyamgj1900/master

Update DG

nonrnp committed:
Grade suggestion no longer take cs/cu mod into calculation

powzx committed:
Merge branch 'master' into branch-Final and resolve merge conflicts in docs/DeveloperGuide.md

nonrnp committed:
Merge pull request #305 from powzx/branch-Final

add instructions for manual testing of planner to DG

nonrnp committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp

shyamgj1900 committed:
Merge pull request #307 from NonRNP/branch-BugFixing

Fix grade suggestion feature with cs/cu mods

nonrnp committed:
Edit cap calculator output message

nonrnp committed:
Edit test cases expected outputs

nonrnp committed:
Add javadoc for isCsCuModule method

nonrnp committed:
Add manual testing instruction for cap calculator

nonrnp committed:
Fix code style

powzx committed:
Merge pull request #308 from NonRNP/branch-UGDG

Add manual testing instruction for cap calculator

powzx committed:
Merge branch 'master' into branch-Final

saravanamani1999 committed:
changed DG timetable add diagram

saravanamani1999 committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-Polish
Merge with upstream

saravanamani1999 committed:
Update DG for timetable add sequence diagram and timetable manual testing

saravanamani1999 committed:
fix checkstyle errors

powzx committed:
Merge pull request #309 from saravanamani1999/branch-Polish

Update UG with new timetable add sequence diagram and add manual testing instructions for timetable

powzx committed:
Merge branch 'master' into branch-Final

powzx committed:
increase plannercommand coverage

powzx committed:
add directorycreatortest

powzx committed:
add some test

powzx committed:
remove directorycreatortest

powzx committed:
fix some doc bugs and prep ug for release

saravanamani1999 committed:
Merge pull request #310 from powzx/branch-Final

Add more JUnit and refine docs

powzx committed:
Merge branch 'master' into branch-Final

saravanamani1999 committed:
update UG and add more Junit tests

giterator committed:
Add Junit test for ModuleListStorage

giterator committed:
Add instructions for manual testing to DG

giterator committed:
Edit heading size for module cap in UG

shyamgj1900 committed:
Merge pull request #311 from saravanamani1999/branch-Polish

update UG and add more Junit tests

shyamgj1900 committed:
Merge branch 'master' into review

nonrnp committed:
Merge pull request #312 from giterator/review

Increase Junit coverage

giterator committed:
Edit PPP

saravanamani1999 committed:
Merge pull request #313 from giterator/review

Edit PPP - giterator

powzx committed:
Merge branch 'master' into branch-Final

powzx committed:
add dummy.txt and start PlannerStorageTest

dummy.txt is a placeholder so that data/ can be pushed to the repo.

powzx committed:
add PlannerStorageTest

powzx committed:
update glossary

nonrnp committed:
Edit logic flow in cap calculator by mc

nonrnp committed:
More JUnit test

nonrnp committed:
Add manual testing instruction for module cap command

nonrnp committed:
Fix code style

powzx committed:
change all profile pics to placeholder

powzx committed:
Merge pull request #314 from NonRNP/branch-UGDG

Add manual instruction for `module cap` command, more JUnit test for cap calculators

powzx committed:
Merge branch 'master' into branch-Final

powzx committed:
cut ppp

saravanamani1999 committed:
add recitation as a lesson type

saravanamani1999 committed:
Merge pull request #315 from powzx/branch-Final

Add PlannerStorageTest and other minor edits

saravanamani1999 committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-Polish
Merge with upstream

saravanamani1999 committed:
update timetable class diagram

powzx committed:
Merge pull request #316 from saravanamani1999/branch-Polish

Added recitation as a lesson type to timetable

powzx committed:
Merge branch 'master' into branch-Final

giterator committed:
Fix DG TOC anchor for module add & delete

saravanamani1999 committed:
Merge pull request #317 from giterator/review

Fix DG table of contents anchor for module add & delete

nonrnp committed:
Add page break

nonrnp committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-UGDG

powzx committed:
Merge pull request #318 from NonRNP/branch-UGDG

Add page break in PPP

powzx committed:
Merge branch 'master' into branch-Final

saravanamani1999 committed:
Add timetable storage tests and fix font sizes for ug/dg

saravanamani1999 committed:
Small edits

saravanamani1999 committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-Polish
Merge with upstream

powzx committed:
Merge pull request #319 from saravanamani1999/branch-Polish

Add timetable storage tests and fix font sizes for ug/dg

powzx committed:
Merge branch 'master' into branch-Final

saravanamani1999 committed:
add images for code snippets with horizontal scroll

saravanamani1999 committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-Polish
Merge with upstream

powzx committed:
Merge pull request #320 from saravanamani1999/branch-Polish

added images for code snippets with horizontal scroll

saravanamani1999 committed:
minor edits to UG and DG

saravanamani1999 committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-Polish
Merge with upstream

powzx committed:
Merge pull request #321 from saravanamani1999/branch-Polish

minor edits to fix UG and DG

powzx committed:
Merge branch 'master' into branch-Final

powzx committed:
resolve hyperlink in table of content

powzx committed:
add all screenshots to ug

saravanamani1999 committed:
Merge pull request #322 from powzx/branch-Final

Add screenshots to all code snippets in UG

powzx committed:
Merge branch 'master' into branch-Final

powzx committed:
delete unused ug screenshots

saravanamani1999 committed:
Merge pull request #324 from powzx/branch-Final

delete all unused ug screenshots

saravanamani1999 committed:
reformat names and add java doc for methods in module details

powzx committed:
Merge branch 'master' into branch-Final

saravanamani1999 committed:
improve code quality

saravanamani1999 committed:
fix checkstyle erros

nonrnp committed:
Move invalidGradeModules attribute into capCalculatorByCode

nonrnp committed:
Revert "Move invalidGradeModules attribute into capCalculatorByCode"

This reverts commit 7c47498b29032ad2124bc5c14840471badbc30ca.

nonrnp committed:
Add condition to check if invalidGradeModules is empty

nonrnp committed:
Merge pull request #325 from saravanamani1999/branch-Polish

Reformat method names in module details and improve code quality in timetable

nonrnp committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-CapCalculator

nonrnp committed:
Add JavaDoc

giterator committed:
Fix PPP typos - giterator

giterator committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into review

shyamgj1900 committed:
Merge pull request #328 from giterator/review

Fix PPP typos - giterator

shyamgj1900 committed:
Merge branch 'master' into branch-CapCalculator

powzx committed:
Merge pull request #326 from NonRNP/branch-CapCalculator

Minor edits for cap calculator

powzx committed:
Merge branch 'master' into branch-Final

powzx committed:
Add javadoc and minor refactors

powzx committed:
fix checkstyle

powzx committed:
improve some messages

powzx committed:
ug edits for consistency

powzx committed:
add page breaks to ug

saravanamani1999 committed:
Merge pull request #329 from powzx/branch-Final

Add javadoc for remaining methods, refactors, and improve error messages

giterator committed:
Add Javadoc for grade class

powzx committed:
Merge pull request #330 from giterator/review

Add Javadoc for grade class

giterator committed:
Moved page break in UG

powzx committed:
Merge pull request #331 from giterator/review

Moved page break in UG

saravanamani1999 committed:
Fix timetable update bug

saravanamani1999 committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W11-1/tp into branch-Polish
Merge with master

powzx committed:
Merge pull request #332 from saravanamani1999/branch-Polish

Fix timetable update bug
